x-pack/plugins/security/server/lib/authorization/disable_ui_capabilities.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mapValues } from 'lodash';
+import { UICapabilities } from 'ui/capabilities';
+import { Actions } from './actions';
+
+export function disableUICapabilitesFactory(server: any, request: any) {
+  const { authorization } = server.plugins.security;
+  const actions: Actions = authorization.actions;
+
+  return async function disableUICapabilities(uiCapabilities: UICapabilities) {
+    const uiActions = Object.entries(uiCapabilities).reduce<string[]>(
+      (acc, [featureId, featureUICapabilities]) => [
+        ...acc,
+        ...Object.keys(featureUICapabilities).map(uiCapability =>
+          actions.ui.get(featureId, uiCapability)
+        ),
+      ],
+      []
+    );
+
+    let checkPrivilegesResponse: any;
```

Comment:
Can we declare a proper type for this? Maybe an interface that's exported out of `check_privileges`/`check_privileges_dynamically`?

Reply:
Great point, will make it so, I forgot that checkPrivilegesDynamically was typescript now. I'm itching to make checkPrivileges itself typescript also.

Reply:
I didn't want to blow the scope out on this PR, but I certainly wouldn't be upset to see `checkPrivileges` migrated!